lib/generator/template_values.rb

```diff
@@ -31,8 +31,40 @@ def cases_require_name
       Files::GeneratorCases.filename(exercise_name)
     end
 
+    def test_cases
+      if test_cases_proc?
+        test_cases_proc.call(canonical_data.to_s)
+      else
+        test_cases_with_index
+      end
+    end
+
+    def test_cases_with_index
+      extract_test_cases.compact.
+        map.with_index {|test, index| test_case_class.new(test.merge('index' => index)) }
+    end
+
+    def extract_test_cases(data = JSON.parse(canonical_data.to_s)['cases'])
+      data.flat_map do |entry|
+        entry.key?('cases') ? extract_test_cases(entry['cases']) : entry
+      end
+    end
+
+    def test_cases_proc?
+      Object.const_defined?(test_cases_procname)
```

Comment:
It was desirable before, for the whole thing to blow up if the test case class did not exist. Since that's no longer the case it suggests to me that `test_cases_proc` needs changing to something like:

```
def test_cases_proc
  Object.const_get(test_cases_procname) if Object.const_defined?(test_cases_procname)
end
```

(This may be moot due to upcoming object extraction.)